etna/models/nn/nbeats/nbeats.py

```diff
@@ -0,0 +1,373 @@
+from etna import SETTINGS
+from etna.distributions import BaseDistribution
+from etna.distributions import FloatDistribution
+from etna.distributions import IntDistribution
+
+if SETTINGS.torch_required:
+    import torch
+
+    from etna.models.base import DeepBaseModel
+    from etna.models.nn.nbeats.metrics import NBeatsLoss
+    from etna.models.nn.nbeats.nets import NBeatsBaseNet
+    from etna.models.nn.nbeats.nets import NBeatsGenericNet
+    from etna.models.nn.nbeats.nets import NBeatsInterpretableNet
+    from etna.models.nn.nbeats.utils import _create_or_update
+    from etna.models.nn.nbeats.utils import prepare_test_batch
+    from etna.models.nn.nbeats.utils import prepare_train_batch
+
+
+class NBeatsBaseModel(DeepBaseModel):
+    """Base class for N-BEATS models."""
+
+    @abstractmethod
+    def __init__(
+        self,
+        net: "NBeatsBaseNet",
+        train_batch_size: int = 1024,
+        test_batch_size: int = 1024,
+        trainer_params: Optional[dict] = None,
+        train_dataloader_params: Optional[dict] = None,
+        test_dataloader_params: Optional[dict] = None,
+        val_dataloader_params: Optional[dict] = None,
+        split_params: Optional[dict] = None,
+        random_state: Optional[int] = None,
+    ):
+        gen_state = np.random.RandomState(seed=random_state)
+        train_collate_fn = partial(
+            prepare_train_batch, input_size=net.input_size, output_size=net.output_size, random_state=gen_state
+        )
+        val_collate_fn = partial(
+            prepare_train_batch, input_size=net.input_size, output_size=net.output_size, random_state=gen_state
+        )
+        test_collate_fn = partial(prepare_test_batch, input_size=net.input_size)
+
+        train_dataloader_params = _create_or_update(
+            param=train_dataloader_params, name="collate_fn", value=train_collate_fn
+        )
+        val_dataloader_params = _create_or_update(param=val_dataloader_params, name="collate_fn", value=val_collate_fn)
+        test_dataloader_params = _create_or_update(
+            param=test_dataloader_params, name="collate_fn", value=test_collate_fn
+        )
+
+        if trainer_params is None or "gradient_clip_val" not in trainer_params:
+            trainer_params = _create_or_update(param=trainer_params, name="gradient_clip_val", value=1.0)
+
+        super().__init__(
+            net=net,
+            encoder_length=net.input_size,
+            decoder_length=net.output_size,
+            train_batch_size=train_batch_size,
+            test_batch_size=test_batch_size,
+            train_dataloader_params=train_dataloader_params,
+            test_dataloader_params=test_dataloader_params,
+            val_dataloader_params=val_dataloader_params,
+            trainer_params=trainer_params,
+            split_params=split_params,
+        )
+
+
+class NBeatsInterpretableModel(NBeatsBaseModel):
+    """Interpretable N-BEATS model.
+
+    Paper: https://arxiv.org/pdf/1905.10437.pdf
```

Comment:
This format renders not very good, you can look at [link](https://deploy-preview-1302--etna-docs.netlify.app/api/etna.models.nn.nbeats.nbeats.nbeatsinterpretablemodel?highlight=nbeatsinterpretablemodel#etna.models.nn.nbeats.nbeats.NBeatsInterpretableModel)

Reply:
Fixed formatting a bit.